vite.config.ts

```diff
@@ -34,10 +34,12 @@ export default defineConfig({
             input: "resources/js/app.tsx",
             refresh: true,
         }),
-        watch({
-            pattern: "app/{Data,Enums}/**/*.php",
-            command: "npm run generate:typescript",
-        }),
+        process.env.NODE_ENV !== "production"
+            ? watch({
+                  pattern: "app/{Data,Enums}/**/*.php",
+                  command: "npm run generate:typescript",
+              })
+            : null,
```

Comment:
Not sure if a `null` plugin will cause any issues, but we could also do it like this:

```suggestion
        ...(process.env.NODE_ENV !== "production"
            ? [watch({
                pattern: "app/{Data,Enums}/**/*.php",
                command: "npm run generate:typescript",
            })]
            : []
        ),
```